### PR TITLE
Add Convert and support conversion from string and boolean

### PR DESCRIPTION
**Pull request by ccoVeille**

### CODE REVIEW

**Comment by ccoVeille** [comment]
This PR is stacked onto #54

- #54 needs to be merged first

File: conversion.go [hunk 1]
```diff
@@ -46,7 +102,41 @@ func convertFromNumber[NumOut Number, NumIn Number](orig NumIn) (converted NumOu
...
 
-// ToInt attempts to convert any [Number] value to an int.
+func convertFromString[NumOut Number](s string) (converted NumOut, err error) {
+	s = strings.TrimSpace(s)
+
+	if b, err := strconv.ParseBool(s); err == nil {
+		if b {
+			return NumOut(1), nil
+		}
+		return NumOut(0), nil
```
Comment (ccoVeille) [inline conversion.go:112]:
```suggestion
		o := 0
		if b {
			o = 1
		}
		return NumOut(o), nil
```

For consistency

File: conversion.go [hunk 2]
```diff
@@ -46,7 +102,41 @@ func convertFromNumber[NumOut Number, NumIn Number](orig NumIn) (converted NumOu
...
+		o, err := strconv.ParseFloat(s, 64)
+		if err != nil {
+			return 0, fmt.Errorf("%w: cannot convert %v to %T", ErrConversionIssue, s, converted)
+		}
+		return convertFromNumber[NumOut](o)
+	}
+
+	if s, found := strings.CutPrefix(s, "-"); found {
+		o, err := strconv.ParseInt(s, 0, 64)
+		if err != nil || o < 0 {
```
Comment (ccoVeille) [inline conversion.go:125]: Add a comment explaining the error returned by ParseInt is irrelevant

Also, the hack of removing the sign to make sure it won't overflow int

Also the check on o < 0, to avoid having string "--123" parsed as 123

File: conversion.go [hunk 3]
```diff
@@ -46,7 +102,41 @@ func convertFromNumber[NumOut Number, NumIn Number](orig NumIn) (converted NumOu
...
+	if strings.Contains(s, ".") {
+		o, err := strconv.ParseFloat(s, 64)
+		if err != nil {
+			return 0, fmt.Errorf("%w: cannot convert %v to %T", ErrConversionIssue, s, converted)
+		}
+		return convertFromNumber[NumOut](o)
+	}
+
+	if s, found := strings.CutPrefix(s, "-"); found {
+		o, err := strconv.ParseInt(s, 0, 64)
```
Comment (ccoVeille) [inline conversion.go:124]: Check if I should not use ParseUint now I removed the sign

Add tests coverage for edge case

Also, things could be rewritten to store the sign and apply it back. If only ParseUint is used
